scripts/src/java/org/oppia/android/scripts/todo/TodoCollector.kt

```diff
@@ -0,0 +1,95 @@
+package org.oppia.android.scripts.todo
+
+import org.oppia.android.scripts.common.RepositoryFile
+import org.oppia.android.scripts.proto.TodoExemptions
+import java.io.File
+import java.io.FileInputStream
+
+/** Helper class to collect all the TODOs in the repository. */
+class TodoCollector {
+  companion object {
+    private val todoDetectorRegex = Regex(
+      pattern = "(?!\".*)\\bTODO\\b(?!.*\")",
+      option = RegexOption.IGNORE_CASE
+    )
+
+    private val todoExemptionTextProto = "scripts/assets/todo_exemptions"
```

Comment:
It seems a bit odd to be doing the exemption filtering here. I'd expect this class just to collect the TODOs & filtering to be done at a higher level since that's an exemption to the enforcement check, not the collection operation.

Reply:
Done.